metanorm/normalizers/ftp.py

```diff
@@ -0,0 +1,153 @@
+"""Normalizer for the interpretation of file name convention"""
+
+import calendar
+import logging
+import re
+import dateutil.parser
+import pythesint as pti
+from dateutil.relativedelta import relativedelta
+from dateutil.tz import tzutc
+from django.contrib.gis.geos.geometry import GEOSGeometry
+
+from .base import BaseMetadataNormalizer
+
+
+LOGGER = logging.getLogger(__name__)
+LOGGER.addHandler(logging.NullHandler())
+
+
+class FTPMetadataNormalizer(BaseMetadataNormalizer):
+    """ Normalizer for hardcoding information for the ftp-derived cases """
+    domain_set = {"anon-ftp.ceda.ac.uk", "ftp.remss.com", "ftp.gportal.jaxa.jp"}
+
+    def match_domain(self, raw_attributes):
+        """ Find the domain in raw_attributes  """
+        if set(['ftp_domain_name']).issubset(raw_attributes.keys()):
+            if raw_attributes['ftp_domain_name'] in self.domain_set:
+                return raw_attributes['ftp_domain_name']
+
+    def get_platform(self, raw_attributes):
+        """ return the corresponding platfrom based on specified ftp source """
+        platform_on_ftp = {"ftp://ftp.remss.com/gmi": 'GPM',
+                           "ftp://anon-ftp.ceda.ac.uk/neodc/esacci/sst/data/CDR_v2/Climatology/L4/v2.1": 'Earth Observation Satellites',
+                           "ftp://ftp.gportal.jaxa.jp/standard/GCOM-W/GCOM-W.AMSR2": 'GCOM-W1'}  # jaxa
+        for ftp_base_dir in platform_on_ftp.keys():
+            if ftp_base_dir in 'ftp://'+raw_attributes['ftp_domain_name'].rstrip('/')+'/'+raw_attributes['ftp_add_and_file_name'].strip('/'):
+                return pti.get_gcmd_platform(platform_on_ftp[ftp_base_dir])
```

Comment:
```suggestion
        for url in urls_platforms.keys():
            if raw_attributes['url'].startswith(url):
                return pti.get_gcmd_platform(urls_platforms[url])
```

Please repeat that logic for `get_instrument`, `get_provider`, `get_dataset_parameter`, `get_geometry`

Reply:
Done in the next commit